src/core/ekat_scalar_traits.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef EKAT_SCALAR_TRAITS_HPP
```

Comment:
This file does not appear as a "renamed" file, since I purged the "invalid" and "quiet_NaN" routines.

The reason is that this file belongs in Core, but as such it cannot depend on Kokkos, and those fcns were marked with KOKKOS_INLINE_FUNCTION (or some flavor of it). In order to avoid circular deps, this class ONLY exposes type traits.

Reply:
Notice that Kokkos offers some nice numeric traits, so we CAN use Kokkos_NumericTraits.hpp for `quiet_NaN_v<T>` or `finite_max_v<T>`. Hence, removing those fcns in this meta-utility is not leaving the user without support for those utilities.